### ISSUE TITLE
Type alias with an import type links to the module, not the referenced type

### ISSUE THREAD

**Issue by blutorange** [TypeStrong/typedoc#2779]
## Search terms

type alias link reference declare module

## Steps to reproduce the bug

Just tried your fix for #2778, works perfectly. Unrelated to that, though, I did notice another strange behavior regarding links to type aliases:

```ts
declare module "bar" {
    export interface Bar {}
}
declare module "foo" {
    export type Foo = import("bar").Bar;
}
```

When you generate the docs, the type alias `Foo` in module `foo` links to the module `bar`, instead of directly to the interface `Bar`.

(Which also results in a missing reference when using `typedoc-plugin-merge-modules`, since, well, the module does not exist anymore as it was merged).

> npx typedoc --entryPoints dist/demo.d.ts --out docs-demo

![image](https://github.com/user-attachments/assets/cd70ae01-3de8-42a2-beb4-09438344526d)

Interestingly, this only happens when using an import type. Importing the symbol first seems to work:

```ts
// demo.d.ts
declare module "bar" {
    export interface Bar {}
}

declare module "foo" {
    import { Bar } from "bar";
    export type Foo = Bar;
}
```

It also seems to happen with `declare module "name"`, but not when using separate files `bar.ts` and `foo.ts`.

## Expected Behavior

TypeDoc should link directly to the referenced type.

## Actual Behavior

TypeDoc links to the the module containing that type.

## Environment

-   Typedoc version: 876c6903d77ba021544f4be0f055fed31688fb4f
-   TypeScript version: 5.6.3
-   Node.js version: v20.18.0
-   OS: Linux `6.8.0-35-generic #35-Ubuntu  x86_64 x86_64 x86_64 GNU/Linux`

**Comment by Gerrit0** [comment]
How on earth has nobody noticed this before... I'm pretty sure it's been there for at least 4 years at this point.

**Comment by blutorange** [comment]
I guess because it only happens with `declare module`, which isn't that common (and was previously converted to namespaces). In our case, after some experimenting, I found creating a merged declaration file with `declare module` to be the best option.

Anyways, I just tried out your change and it works as expected. With that, I think I now also have a working setup for generating a nice looking documentation for primefaces/primefaces#12717 Thanks again for fixing this so quickly.

![image](https://github.com/user-attachments/assets/1ce9012f-c149-46db-bbbc-4551268f23f5)
